### ISSUE TITLE
[lib.cuckoo.core.plugins] ERROR: Failed to run the processing module "NetworkAnalysis"

### ISSUE THREAD

**Issue by jeffhammett** [cuckoobox/cuckoo#546]
I just setup Cuckoo on Ubuntu 14.04 LTS. Everything seems to be working except for generating pcaps. I downloaded Cuckoo from cuckoosandbox.org, I am getting the following error on all files I analyze, any ideas on how to fix?

2015-05-11 16:16:34,688 [lib.cuckoo.core.plugins] ERROR: Failed to run the processing module "NetworkAnalysis":
Traceback (most recent call last):
  File "/home/user/bin/cuckoo/lib/cuckoo/core/plugins.py", line 186, in process
    data = current.run()
  File "/home/user/bin/cuckoo/modules/processing/network.py", line 593, in run
    sort_pcap(self.pcap_path, sorted_path)
  File "/home/user/bin/cuckoo/modules/processing/network.py", line 708, in sort_pcap
    batch_sort(inc, outpath, output_class=lambda path: SortCap(path, linktype=inc.linktype))
  File "/home/user/bin/cuckoo/modules/processing/network.py", line 635, in batch_sort
    current_chunk = list(islice(input_iterator,buffer_size))
  File "/home/user/bin/cuckoo/modules/processing/network.py", line 677, in __iter__
    self.fd = dpkt.pcap.Reader(open(self.name, "rb"))
IOError: [Errno 2] No such file or directory: '/home/user/bin/cuckoo/storage/analyses/1/dump.pcap'
2015-05-11 16:16:37,317 [lib.cuckoo.core.scheduler] INFO: Task #1: reports generation completed (path=/home/user/bin/cuckoo/storage/analyses/1)
2015-05-11 16:16:37,382 [lib.cuckoo.core.scheduler] INFO: Task #1: analysis procedure completed

**Comment by tysonmax20042003** [comment]
This sounds like the same issue as myself. 

Open issue: Analysis not starting on Proxmox KVM #519
https://github.com/cuckoobox/cuckoo/issues/519

Cuckoo Sandbox 1.3-dev

2015-05-05 22:28:19,682 [lib.cuckoo.core.scheduler] ERROR: Failure in AnalysisManager.run
Traceback (most recent call last):
File "/home/cuckoo/bin/cuckoo/lib/cuckoo/core/scheduler.py", line 363, in run
success = self.launch_analysis()
File "/home/cuckoo/bin/cuckoo/lib/cuckoo/core/scheduler.py", line 281, in launch_analysis
machinery.stop(self.machine.label)
File "/home/cuckoo/bin/cuckoo/modules/machinery/physical.py", line 88, in stop
shutdown = subprocess.Popen(args, stdout=subprocess.PIPE)
File "/usr/lib/python2.7/subprocess.py", line 710, in init
errread, errwrite)
File "/usr/lib/python2.7/subprocess.py", line 1327, in execute_child
raise child_exception
OSError: [Errno 2] No such file or directory

**Comment by tysonmax20042003** [comment]
Try running cuckoo in debug mode to see if you get more details on the fails. IE: python ./cuckoo.py -d

**Comment by KillerInstinct** [comment]
@tysonmax20042003 these errors appear to be unreleated from what I can tell.

@jeffhammett I believe this is due to a configuration issue, or not setting up tcpdump properly. Please check that you have allowed non-root users the ability to run tcpdump, and that your conf/auxiliary.conf file properly reflects your configured interface.

To allow non-root users to run tcpdump, view the following documentation:
https://cuckoo.readthedocs.org/en/latest/installation/host/requirements/#installing-tcpdump

**Comment by jeffhammett** [comment]
As far as I can tell tcpdump is setup for non-root users, and my interface is setup correctly in the conf files. Below is output of getcap /usr/sbin/tcpdump, ifconfig and both my cuckoo.conf and auxilliary.conf

    $ getcap /usr/sbin/tcpdump
    /usr/sbin/tcpdump = cap_net_admin,cap_net_raw+eip
    
    $ ifconfig
    eth0      Link encap:Ethernet  HWaddr MAC  
              inet addr:192.168.1.129  Bcast:192.168.1.255  Mask:255.255.255.0
              inet6 addr: fe80::d685:64ff:fe14:533b/64 Scope:Link
              UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
              RX packets:1155048 errors:0 dropped:0 overruns:0 frame:0
              TX packets:501566 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:1000 
              RX bytes:1513177426 (1.5 GB)  TX bytes:38896036 (38.8 MB)
              Interrupt:19 Memory:f3000000-f3020000 
    
    lo        Link encap:Local Loopback  
              inet addr:127.0.0.1  Mask:255.0.0.0
              inet6 addr: ::1/128 Scope:Host
              UP LOOPBACK RUNNING  MTU:65536  Metric:1
              RX packets:3760 errors:0 dropped:0 overruns:0 frame:0
              TX packets:3760 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:0 
              RX bytes:6151166 (6.1 MB)  TX bytes:6151166 (6.1 MB)
    
    vboxnet0  Link encap:Ethernet  HWaddr 0a:00:27:00:00:00  
              inet addr:192.168.240.1  Bcast:192.168.240.255  Mask:255.255.255.0
              inet6 addr: fe80::800:27ff:fe00:0/64 Scope:Link
              UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
              RX packets:0 errors:0 dropped:0 overruns:0 frame:0
              TX packets:539 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:1000 
              RX bytes:0 (0.0 B)  TX bytes:78349 (78.3 KB)
    
    virbr0    Link encap:Ethernet  HWaddr 5e:c6:51:fe:f4:cb  
              inet addr:192.168.122.1  Bcast:192.168.122.255  Mask:255.255.255.0
              UP BROADCAST MULTICAST  MTU:1500  Metric:1
              RX packets:0 errors:0 dropped:0 overruns:0 frame:0
              TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
              collisions:0 txqueuelen:0 
              RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)
    
    $ cat cuckoo.conf 
    [cuckoo]
    # Enable or disable startup version check. When enabled, Cuckoo will connect
    # to a remote location to verify whether the running version is the latest
    # one available.
    version_check = on
    
    # If turned on, Cuckoo will delete the original file after its analysis
    # has been completed.
    delete_original = off
    
    # If turned on, Cuckoo will delete the copy of the original file in the
    # local binaries repository after the analysis has finished. (On *nix this
    # will also invalidate the file called "binary" in each analysis directory,
    # as this is a symlink.)
    delete_bin_copy = off
    
    # Specify the name of the machinery module to use, this module will
    # define the interaction between Cuckoo and your virtualization software
    # of choice.
    machinery = virtualbox
    
    # Enable creation of memory dump of the analysis machine before shutting
    # down. Even if turned off, this functionality can also be enabled at
    # submission. Currently available for: VirtualBox and libvirt modules (KVM).
    memory_dump = on
    
    # When the timeout of an analysis is hit, the VM is just killed by default.
    # For some long-running setups it might be interesting to terminate the
    # moinitored processes before killing the VM so that connections are closed.
    terminate_processes = off
    
    # Enable automatically re-schedule of "broken" tasks each startup.
    # Each task found in status "processing" is re-queued for analysis.
    reschedule = off
    
    # Enable processing of results within the main cuckoo process.
    # This is the default behavior but can be switched off for setups that
    #  require high stability and process the results in a separate task.
    process_results = on
    
    # Limit the amount of analysis jobs a Cuckoo process goes through.
    # This can be used together with a watchdog to mitigate risk of memory leaks.
    max_analysis_count = 0
    
    # Limit the number of concurrently executing analysis machines.
    # This may be useful on systems with limited resources.
    # Set to 0 to disable any limits.
    max_machines_count = 0
    
    # Minimum amount of free space (in MB) available before starting a new task. 
    # This tries to avoid failing an analysis because the reports can't be written 
    # due out-of-diskspace errors. Setting this value to 0 disables the check.
    # (Note: this feature is currently not supported under Windows.)
    freespace = 64
    
    # Temporary directory containing the files uploaded through Cuckoo interfaces
    # (web.py, api.py, Django web interface).
    tmppath = /tmp
    
    [resultserver]
    # The Result Server is used to receive in real time the behavioral logs
    # produced by the analyzer.
    # Specify the IP address of the host. The analysis machines should be able
    # to contact the host through such address, so make sure it's valid.
    # NOTE: if you set resultserver IP to 0.0.0.0 you have to set the option
    # `resultserver_ip` for all your virtual machines in machinery configuration.
    ip = 192.168.240.1
    
    # Specify a port number to bind the result server on.
    port = 2042
    
    # Should the server write the legacy CSV format?
    # (if you have any custom processing on those, switch this on)
    store_csvs = off
    
    # Maximum size of uploaded files from VM (screenshots, dropped files, log)
    # The value is expressed in bytes, by default 10Mb.
    upload_max_size = 10485760
    
    [processing]
    # Set the maximum size of analyses generated files to process. This is used
    # to avoid the processing of big files which may take a lot of processing
    # time. The value is expressed in bytes, by default 100Mb.
    analysis_size_limit = 104857600
    
    # Enable or disable DNS lookups.
    resolve_dns = on
    
    # Enable PCAP sorting, needed for the connection content view in the web interface.
    sort_pcap = on
    
    [database]
    # Specify the database connection string.
    # Examples, see documentation for more:
    # sqlite:///foo.db
    # postgresql://foo:bar@localhost:5432/mydatabase
    # mysql://foo:bar@localhost/mydatabase
    # If empty, default is a SQLite in db/cuckoo.db.
    connection = mysql://cuckoo:password@localhost/cuckoo
    
    # Database connection timeout in seconds.
    # If empty, default is set to 60 seconds.
    timeout =
    
    [timeouts]
    # Set the default analysis timeout expressed in seconds. This value will be
    # used to define after how many seconds the analysis will terminate unless
    # otherwise specified at submission.
    default = 120
    
    # Set the critical timeout expressed in seconds. After this timeout is hit
    # Cuckoo will consider the analysis failed and it will shutdown the machine
    # no matter what. When this happens the analysis results will most likely
    # be lost. Make sure to have a critical timeout greater than the
    # default timeout.
    critical = 600
    
    # Maximum time to wait for virtual machine status change. For example when
    # shutting down a vm. Default is 300 seconds.
    vm_state = 300
    
    $ cat auxiliary.conf 
    [sniffer]
    # Enable or disable the use of an external sniffer (tcpdump) [yes/no].
    enabled = yes
    
    # Specify the path to your local installation of tcpdump. Make sure this
    # path is correct.
    tcpdump = /usr/sbin/tcpdump
    
    # Specify the network interface name on which tcpdump should monitor the
    # traffic. Make sure the interface is active.
    interface = vboxnet0
    
    # Specify a Berkeley packet filter to pass to tcpdump.
    # bpf = not arp

**Comment by jeffhammett** [comment]
I also tested by running

    $ tcpdump -i vboxnet0 

in Ubuntu and browsing in IE on the virtual machine and packets are captured just fine.

**Comment by jhg** [comment]
I has same error and I try all for configure tcpdump and with `tcpdump -i vboxnet0 work`:
>2015-06-03 16:24:52,513 [lib.cuckoo.core.plugins] ERROR: Failed to run the processing module "NetworkAnalysis":
>Traceback (most recent call last):
>  File "/opt/cuckoo/lib/cuckoo/core/plugins.py", line 186, in process
    data = current.run()
>  File "/opt/cuckoo/modules/processing/network.py", line 593, in run
    sort_pcap(self.pcap_path, sorted_path)
>  File "/opt/cuckoo/modules/processing/network.py", line 708, in sort_pcap
    batch_sort(inc, outpath, output_class=lambda path: SortCap(path, linktype=inc.linktype))
>  File "/opt/cuckoo/modules/processing/network.py", line 635, in batch_sort
    current_chunk = list(islice(input_iterator,buffer_size))
>  File "/opt/cuckoo/modules/processing/network.py", line 677, in __iter__
    self.fd = dpkt.pcap.Reader(open(self.name, "rb"))
>IOError: [Errno 2] No such file or directory: '/opt/cuckoo/storage/analyses/15/dump.pcap'

**Comment by jbremer** [comment]
Sorry guys, I only see this issue now. The following commit from roughly a month ago in the `master` branch fixes it: https://github.com/cuckoobox/cuckoo/commit/cafd67e1ec80b2b358a4e1b16cc65e471e1d6702. Please upgrade or manually apply it.. that should do the trick.

**Comment by jbremer** [comment]
Another work around is disabling `sort_pcap` by changing to `sort_pcap = no` in `conf/cuckoo.conf`.

**Comment by jhg** [comment]
Thanks @jbremer

**Comment by botherder** [comment]
Closing as fixed.
